source/family/freescale/target_reset_mimxrt.c

```diff
@@ -47,14 +47,21 @@ static void prerun_target_config(void)
 
 static uint8_t validate_bin_nvic(const uint8_t *buf)
 {
+    #include "target_config.h"
+    extern target_cfg_t target_device ;
+
     if(buf[0] == 'F' && buf[1] == 'C' && buf[2] == 'F' && buf[3] == 'B')
     {
+        target_device.flash_regions[0].start = 0x30000400;
         return 1;
     }
-    else
+    else if(buf[0] == 0xFF && buf[1] == 0xFF && buf[2] == 0xFF && buf[3] == 0xFF)
     {
-        return 0;
+        target_device.flash_regions[0].start = 0x30000000;
+        return 1;
     }
+
+    return 0;
```

Comment:
Nice hack! Would you mind adding some comments describing what this code does so we don't forget in the future? 😄

Reply:
From what I understand, if buf is a FlexSPI Config block (the "FCFB"/0x42464346 test), it moves the flash start address;
but I have no idea what the test against 0xFFFFFFFF part does; I see some other platforms have a "cookieList"